bottlerocket/samples/scripts/create_sample_file.sh

```diff
@@ -0,0 +1,299 @@
+#!/bin/bash
+
+function create_ecs_migration_test() {
```

Comment:
Is it possible to deduplicate any of the code? I had originally envisioned all of these env variables being set at the top of the Makefile so a user could overwrite them, and then the target would just do the variable substation and run the yaml.

Reply:
That was my initial plan as well, but I spent the better part of yesterday trying to do just that and I was not able to figure out variable substitution inside a Makefile target. Both `eval` and `envsubst` do not work.

Reply:
In the makefile you should be able to do something like:

```bash
VARIANT=aws-k8s-1.24 ARCHITECTURE=x86_64 scripts/create_sample_file.sh sonobuoy-migration eks
```

Reply:
^ That would still require having the script, which is fine by me, but I'm not sure if that aligns with ecpullen's vision. So now I'm thinking: export all variables at the top of the Makefile if not already set, and inside the target, just call the script with the two arguments, which does the variable substitution and writes the output file. The output file location is echo'd, so the Makefile target can run the test. @ecpullen Thoughts?

Reply:
Ideally, the logic would be unified such that when a new sample is added, the new sample could be added to a `samples` list at the top of the Makefile (similar to how we add new bottlerocket agents) with nothing else (except additional env variables that aren't already in the Makefile). I am open to the implementation details if that is the end result.

Reply:
The problem is that as far as I can tell, each Makefile variable needs to be passed in explicitly to the create_script command. I changed the Makefile so it now contains all variables and the script doesn't have any duplicated code anymore, but it's still not exactly what you want. For example, it doesn't seem possible to set VARIANT if not already set (VARIANT ?= ...) inside of a make target, so the only option I can think of is to set VARIANT inside the bash script. Consequence of this is that if we add a new sample file, we'll have to add a case to the switch statement in the bash script as well as add a new target in the makefile that looks like the other targets currently in there.